index.js

```diff
@@ -57,6 +60,12 @@ Some.prototype.flatMap = function(func) {
     return func(this._value);
 };
 
+Some.prototype.filter = function(f) {
+    return this.flatMap(function(x) {
+        return f(x) ? exports.some(x) : exports.none;
+    });
```

Comment:
This seems a little overcomplicated. Wouldn't the following accomplish the same thing?

```javascript
Some.prototype.filter = function(predicate) {
    return predicate(this._value) ? this : none;
};
```